tests/test_chiral_utils.py

```diff
@@ -0,0 +1,168 @@
+
+
+def test_setup_chiral_atom_restraints():
+    """On a methane conformer, assert that permuting coordinates or permuting restr_idxs
+    both independently toggle the chiral restraint"""
+    mol = Chem.MolFromMolBlock(
+        """
+  Mrv2202 06072215563D
+
+  5  4  0  0  0  0            999 V2000
+    0.0000    0.0000    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
+   -0.3633   -0.5138    0.8900 H   0  0  0  0  0  0  0  0  0  0  0  0
+    1.0900    0.0000    0.0000 H   0  0  0  0  0  0  0  0  0  0  0  0
+   -0.3633    1.0277    0.0000 H   0  0  0  0  0  0  0  0  0  0  0  0
+   -0.3633   -0.5138   -0.8900 H   0  0  0  0  0  0  0  0  0  0  0  0
+  1  2  1  0  0  0  0
+  1  3  1  0  0  0  0
+  1  4  1  0  0  0  0
+  1  5  1  0  0  0  0
+M  END
+$$$$""",
+        removeHs=False,
+    )
+
+    # needs to be batched in order for jax to play nicely
+    x0 = utils.get_romol_conf(mol)
+    normal_restr_idxs = chiral_utils.setup_chiral_atom_restraints(mol, x0, 0)
+
+    x0_inverted = x0[[0, 2, 1, 3, 4]]  # swap two atoms
+    inverted_restr_idxs = chiral_utils.setup_chiral_atom_restraints(mol, x0_inverted, 0)
+
+    # check the sign of the resulting idxs
+    k = 1000.0
+    assert np.all(np.asarray(U_chiral_atom_batch(x0, normal_restr_idxs, k)) == 0)
+    assert np.all(np.asarray(U_chiral_atom_batch(x0, inverted_restr_idxs, k)) > 0)
+    assert np.all(np.asarray(U_chiral_atom_batch(x0_inverted, normal_restr_idxs, k)) > 0)
+    assert np.all(np.asarray(U_chiral_atom_batch(x0_inverted, inverted_restr_idxs, k)) == 0)
+
+
+def test_setup_chiral_bond_restraints():
+    """On a 'Cl/C(F)=N/F' conformer, assert that flipping a dihedral angle or permuting restr_idxs
+    both independently toggle the chiral bond restraint"""
+
+    mol_cis = Chem.MolFromSmiles(r"Cl\C(F)=N/F")
+    mol_trans = Chem.MolFromSmiles(r"Cl\C(F)=N\F")
+
+    AllChem.EmbedMolecule(mol_cis)
+    AllChem.EmbedMolecule(mol_trans)
+
+    # needs to be batched in order for jax to play nicely
+    x0_cis = utils.get_romol_conf(mol_cis)
+    x0_trans = utils.get_romol_conf(mol_trans)
+    src_atom = 1
+    dst_atom = 3
+    normal_restr_idxs, signs = chiral_utils.setup_chiral_bond_restraints(mol_cis, x0_cis, src_atom, dst_atom)
+
+    inverted_restr_idxs, inverted_signs = chiral_utils.setup_chiral_bond_restraints(
+        mol_trans, x0_trans, src_atom, dst_atom
+    )
+    k = 1000.0
+
+    assert np.all(np.asarray(U_chiral_bond_batch(x0_cis, normal_restr_idxs, k, signs)) == 0)
+    assert np.all(np.asarray(U_chiral_bond_batch(x0_cis, inverted_restr_idxs, k, inverted_signs)) > 0)
+    assert np.all(np.asarray(U_chiral_bond_batch(x0_trans, normal_restr_idxs, k, signs)) > 0)
+    assert np.all(np.asarray(U_chiral_bond_batch(x0_trans, inverted_restr_idxs, k, inverted_signs)) == 0)
+
+
+def test_find_chiral_atoms():
+
+    mol = Chem.MolFromSmiles(r"FC(Cl)Br")
+    res = chiral_utils.find_chiral_atoms(mol)
+    assert res == set([1])
```

Comment:
This one relies on `mol.GetAtomWithIdx(1).GetNumImplicitHs() == 1`, would be slightly clearer if the hydrogen weren't implicit, since we generally will only be applying this method to rdmols with explicit hydrogens?

```suggestion
    mol = Chem.AddHs(Chem.MolFromSmiles(r"FC(Cl)Br"))
    res = chiral_utils.find_chiral_atoms(mol)
    assert res == set([1])
```

Reply:
Agreed - sorry I missed the implicit H here!